DeveloperTools/Controllers/LocalObjectCacheController.cs

```diff
@@ -0,0 +1,125 @@
+    {
+        private readonly ISynchronizedObjectInstanceCache _cache;
+
+        public LocalObjectCacheController(ISynchronizedObjectInstanceCache cache)
+        {
+            _cache = cache;
+        }
+
+        public ActionResult Index(string FilteredBy, bool os = false)
+        {
+            return View(PrepareViewModel(FilteredBy, os));
+        }
+
+        [HttpParamAction]
+        public ActionResult RemoveLocalCache(string[] cacheKeys, bool os)
+        {
+            if(cacheKeys != null)
+            {
+                foreach (string key in cacheKeys)
+                {
+                    _cache.RemoveLocal(key);
+                }
+            }
+
+            return RedirectToAction("Index", new RouteValueDictionary(new { os }));
+        }
+
+        [HttpParamAction]
+        public ActionResult RemoveLocalRemoteCache(string[] cacheKeys, bool os)
+        {
+            if(cacheKeys != null)
+            {
+                foreach (string key in cacheKeys)
+                {
+                    _cache.RemoveLocal(key);
+                    _cache.RemoveRemote(key);
+                }
+            }
+
+            return RedirectToAction("Index", new RouteValueDictionary(new { os }));
+        }
+
+        [HttpParamAction]
+        public ActionResult ViewObjectSize()
+        {
+            return RedirectToAction("Index", new RouteValueDictionary(new { os = true }));
+        }
+
+        private LocalObjectCache PrepareViewModel(string FilteredBy, bool viewObjectSize)
+        {
+            var model = new LocalObjectCache();
+
+            var cachedEntries = HttpContext.Cache.Cast<DictionaryEntry>();
+
+            switch (FilteredBy)
+            {
+                case "pages":
+                    model.CachedItems = ConvertToListItem(cachedEntries.Where(item => item.Value is PageData), viewObjectSize);
+                    break;
+                case "content":
+                    model.CachedItems = ConvertToListItem(cachedEntries.Where(item => item.Value is IContent), viewObjectSize);
+                    break;
+                default:
+                    model.CachedItems = ConvertToListItem(cachedEntries, viewObjectSize);
+                    break;
+            }
+
+            model.FilteredBy = FilteredBy;
+            model.Choices = new[]
+                            {
+                                new SelectListItem { Text = "All Cached Objects", Value = "all" },
+                                new SelectListItem { Text = "Any Content", Value = "content" },
+                                new SelectListItem { Text = "Pages Only", Value = "pages" }
+                            };
+
+            model.ViewObjectSize = viewObjectSize;
+            return model;
+        }
+
+        private IEnumerable<LocalObjectCacheItem> ConvertToListItem(IEnumerable<DictionaryEntry> cachedEntries, bool viewObjectSize) =>
+            cachedEntries.Select(e => new LocalObjectCacheItem

```

Comment:
I think just adding a Take(10,000) here would protect against issues with really large caches until server side paging is in place.  This code otherwise have the potential of running a server out of memory.